artemis/module_base.py

```diff
@@ -265,6 +293,10 @@ def _take_and_lock_tasks(self, num_tasks: int) -> Tuple[List[Task], List[Optiona
         finally:
             self.taking_tasks_from_queue_lock.release()
 
+        if len(tasks) < num_tasks:
+            self.queue_id = 0
+            self.queue_position = 0
+            self.queue_location_timestamp = time.time()
```

Comment:
same comment as above, best if timestamp is the same in this funcblock

Reply:
I don't see any reason to do that, I want to update the timestamp simply to current time, no problem if the two time..time() calls diverge, even better to have the current one if we spent some time in the function